compiler/src/iree/compiler/Dialect/VM/IR/BUILD.bazel

```diff
@@ -53,6 +53,10 @@ iree_compiler_cc_library(
         "VMTraits.h",
         "VMTypes.h",
     ],
+    copts = [
+        # GCC warning from mlir DialectImplementation.h
+        "-Wno-uninitialized",
+    ],
```

Comment:
Update the build setting to only turn off the warning on GCC w.r.t the specific header. 

Filed an LLVM issue and I will try to propose the PR to fix DialectImplementation.h upstream.

Reply:
Ick, `#pragma`s :P not pretty, but fine to proceed with while the warnings are fixed upstream IMO.

Reply:
Not sure how to fix DenseMap.h since it was introduced in https://github.com/llvm/llvm-project/commit/20dd838ae532b70115a8cb84bb9e6857cad6015c back in 2012. DialectImplementation.h is something we can fix more easily.